component/datamodel/util/file-utility.js

```diff
@@ -111,3 +112,12 @@ function readDataSource(workspace, cb) {
     cb(null, data);
   });
 }
+
+function readMiddleware(workspace, cb) {
+  const filePath = workspace.getMiddlewareFilePath();
+  fs.readJson(filePath, function(err, data) {
+    if (err) return err;
```

Comment:
this seems like an odd pattern. That means usage would look like:

```
var err = readMiddleWare('ws', function(iDontKnow, data) {
  // stuff that will never use the first param of the callback function
});
if (err) (console.error('there was an error outside of the callback');
```

Seems like it should call the callback instead of returning err. Kind of weird that the return of the function is true if there is an error and undefined if actually passes.